integration_tests/client/datatype/test_data_generation.py

```diff
@@ -313,7 +312,7 @@ def test_generate_segmentation_data(
 
     for image in dataset.get_datums():
         uid = image.uid
-        sample_gt = dataset.get_groundtruth(uid)
+        sample_gt = dataset.get_groundtruth(uid)  # type: ignore - issue #604
```

Comment:
what's causing there to be a type issue on this line with these changes?

Reply:
Removed type ignore. There is red highlighting but it passes pre-commit.

Reply:
I thought this was to be the last filtering PR, so I was expecting issues like this one to be fixed. I'm ok with pushing this into another PR.

Reply:
It is the last one to work on filter structure. These type ignores are related to pyright issues that only affect json generation in the client.